koel/core
# Upgrade webpack to latest

webpack's (major) changelog: https://github.com/webpack/webpack/releases/tag/v4.0.0
file-loader (major) changelog: https://github.com/webpack-contrib/file-loader/blob/master/CHANGELOG.md

Paths to images have been changed to relative paths. Previous paths
resulted in errors like "Cannot find module 'null'".

I have not upgraded sass-loader and vue-loader yet, let's do this in a
dedicated contribution.

How to test:
- You will need to checkout the pull request for koel (server) to have
  the matching laravel-mix version.
- yarn run production (at koel server root) should succeed and build the
  assets

Note to reviewer:
- I removed node 8 from targets of travis CI. file-loader v5.0.0 requires nodejs >= 10.13.0 [see changelog](https://github.com/webpack-contrib/file-loader/blob/master/CHANGELOG.md#500-2019-11-22)
- I have updated snapshots but it looks like some changes were not due to this contribution.
For example `js/tests/components/layout/main-wrapper/index.spec.js`. I tried to stub manually the "main-content.vue" component because it kept changing (greetings are randomized) and breaking the snapshot.
